lib/Frontend/nvqpp/ConvertExpr.cpp

```diff
@@ -1837,6 +1837,42 @@ bool QuakeBridgeVisitor::VisitInitListExpr(clang::InitListExpr *x) {
     // Pass initialization list with one member as a Ref.
     return pushValue(last[0]);
   }
+
+  // Check if all values here are Integer or Float Type
+  bool isAllIntOrAllFloat = [&]() {
+    for (auto v : last)
+      if (!v.getType().isIntOrFloat())
+        return false;
+    return true;
+  }();
+
+  // If these are integers or floats, then let's allocate
+  // some memory and store them there.
+  if (isAllIntOrAllFloat) {
+    // Clear the types for the init expr
+    typeStack.clear();
+
+    // This is a initlist on ints or floats, get which one
+    Type dataType = last.front().getType();
```

Comment:
The loop above doesn't preclude having some int's and some float's. Does something else prevent that? If not, then looking at the first element may not be enough.

Reply:
This is a good point, the code here may be general already for type T in std::vector<T>. I will see if I can remove the check on ints/floats.

Reply:
@schweitzpgi Do you think I can remove the check on ints/ floats here and leave this code as the default case for visiting an InitListExpr?

Reply:
I don't think we need to check.

clang should make sure all the values in `last` conform to the element type of the `std::vector`. (Although we could be a bit paranoid for the moment and assert this assumption to be safe.)